lib/server/query.js

```diff
@@ -62,7 +63,15 @@ function enforceDateFilter (query, opts) {
       let dateString = dateValue[key];
       if (isNaN(dateString)) {
         dateString = dateString.replace(' ', '+'); // some clients don't excape the plus
-        dateValue[key] = new Date(dateString).toISOString();
+
+        const validDate = moment(dateString).isValid();
+
+        if (!valid) {
```

Comment:
I think you meant to check if validDate is true.

Reply:
Oops, had pushed the working code to only my personal fork, apologies